### DISCUSSION TITLE
进阶任务集合～

### DISCUSSION THREAD

**Discussion by chuntaojun** [polarismesh/polaris#357] [Show and tell]
## 如何发现进阶任务呢？

可以在[**polarismesh**](https://github.com/polarismesh)这个组织下相关项目的issue列表中，查看打上了**good advanced issue**标签的

![企业微信截图_bb3448a3-bb92-4a8a-b592-066d0284dd09](https://user-images.githubusercontent.com/23445172/165248154-b85b13d8-861a-4b4d-b4a4-b617cc51667e.png)

## 如何认领任务

在对应的任务中回复 assigne me 即可

**Comment by chuntaojun** [comment]


**Reply by chuntaojun** [reply]
@[shichaoyuan](https://github.com/shichaoyuan) claimed

**Reply by chuntaojun** [reply]
this task finished

**Comment by chuntaojun** [comment]
# Polaris控制台支持国际化

polaris-console：https://github.com/polarismesh/polaris-console/issues/46

建议实现思路

- 实现中文、英文通过前端的一个按钮点击进行切换
- 参考前端国际化 i18n 的相关组件

**Reply by Skyenought** [reply]
assigne me

**Comment by chuntaojun** [comment]
# 增强北极星的命令行运维能力

polaris-server：https://github.com/polarismesh/polaris/issues/150

建议实现思路

- 实现 polarisctl 命令行工具
- 该工具可以管理多个polaris集群，可以参考 kubectl 的 context 机制
- 可以使用该 polarisctl 调用 polaris 的运维接口
- 可以使用类似 yaml 文件的形式，polarisctl create -f ... 创建 polaris 相关资源
  - 命名空间
  - 服务
  - 服务实例
  - 服务治理规则
  - 配置分组
  - 配置文件
- 可以使用该工具修改 polaris 的鉴权中与 token 加密的 salt
  - 数据库中 token 字段的重新加密生成
  - 配置文件中 salt 值的调整

**Comment by chuntaojun** [comment]
# 为 Polaris 实现一个内嵌的分布式存储插件

polaris-server：https://github.com/polarismesh/polaris/issues/145

建议实现思路

- 确定内嵌分布式存储的存储组件选型
- 确定数据共识协议的选型
  - Raft：强一致
  - Gossip：最终一致（可以适当对 Gossip 进行变种）
  - Eureka：最终一致（可以考虑和Gossip结合一下）
- 节点间如何自动发现对方
  - 初期可以考虑使用指定的配置文件，该文件指定了成员列表
  - 后面可以考虑怎么利用类似广播的方式，去发现集群间的节点
- 实现store接口
- 设计测试用例，以及考虑对接混沌工程，确保该分布式存储组件的数据一致、容错机制是ok的
- 可观测性如何设计
- 可运维性如何设计

**Reply by shichaoyuan** [reply]
这种模式的话，cache模块就不需要了，整体变动还挺大

**Reply by chuntaojun** [reply]
> 这种模式的话，cache模块就不需要了，整体变动还挺大

cache 模块还是需要的，读缓存，写store，cache模块是为了读加速的

**Reply by shichaoyuan** [reply]
> > 这种模式的话，cache模块就不需要了，整体变动还挺大
> 
> cache 模块还是需要的，读缓存，写store，cache模块是为了读加速的

可能是我理解偏了。我想的是内嵌存储插件依赖共识协议同步数据，这样每个节点都有完整的数据，这样 cache+store 二合一。

**Reply by chuntaojun** [reply]
> > > 这种模式的话，cache模块就不需要了，整体变动还挺大
> > 
> > 
> > cache 模块还是需要的，读缓存，写store，cache模块是为了读加速的
> 
> 可能是我理解偏了。我想的是内嵌存储插件依赖共识协议同步数据，这样每个节点都有完整的数据，这样 cache+store 二合一。

store层是插件化的，不能说因为加上了这个共识之后，使得cache的逻辑就跟这个内嵌存储的实现耦合在一起，cache层就让他保持为一个简单的数据缓存以及加速读操作，store层只需要看看怎么去能够让cache获取到增量的数据变化即可

**Reply by andrewshan** [reply]
cache层还是要有，共识算法主要是解决的是数据存储的问题。
比如eureka，数据的写和读也是分开2个Map，做了冗余的。

**Reply by bingxindan** [reply]
第一次接触这个，可以参与搞吗？

**Reply by chuntaojun** [reply]
> 第一次接触这个，可以参与搞吗？

可以的

**Reply by bingxindan** [reply]
assigne me

**Reply by bingxindan** [reply]
咱们有交流的群吗？能拉我下吗？

**Reply by chuntaojun** [reply]
> 咱们有交流的群吗？能拉我下吗？

可以直接加我wechat: LCT1069284099

**Reply by chuntaojun** [reply]
> assigne me

在对应的 issue 下回复下？

**Comment by chuntaojun** [comment]
# 添加ConfigChain组件，为配置中心模块提供更多能力

polaris-server：https://github.com/polarismesh/polaris/issues/384

目标

- 在 apiserver -> configservice -> store 链路中的 configservice -> store 中，添加 ConfigChain 插件
- ConfigChain 内有各种 ConfigFileHandler 插件
  - 根据定义的 ConfigFileHandler 插件，预期实现以下功能
    - 配置加密能力的 handler
    - 配置发布审批

建议实现思路

- 了解北极星配置中心的整体逻辑
- 设计ConfigChain插件
- 利用现有的ConfigFile相关属性，设计好EncryptConfigFileHandler插件

**Reply by shichaoyuan** [reply]
这个是想对 service 层的 api 做处理链？还是 store 层的 api？

**Reply by chuntaojun** [reply]
> service 层的 api

这个等我完善下相关信息，目的是 service 到 store

**Reply by shichaoyuan** [reply]
assigne me

**Reply by chuntaojun** [reply]
> 建议实现思路
> 
> * 了解北极星配置中心的整体逻辑
> * 设计ConfigChain插件
> * 利用现有的ConfigFile相关属性，设计好EncryptConfigFileHandler插件

这个建议可以先暂缓一下，因为可能配置中心有的逻辑要先调整下

**Reply by shichaoyuan** [reply]
好的

**Comment by chuntaojun** [comment]
# polaris-cpp 支持配置中心能力

polaris-cpp：https://github.com/polarismesh/polaris-cpp/issues/30

建议实现思路：参考 polaris-java 的配置中心接口实现

**Comment by chuntaojun** [comment]
# 北极星服务端支持查询服务之间的订阅关系

polaris-server：https://github.com/polarismesh/polaris/issues/373

目标

- 北极星前端支持按照服务查询服务的订阅关系、当前某个IP订阅服务的版本信息

建议实现思路

- 了解北极星服务端整体设计
- 了解 ReportClient 接口的设计
- 尝试通过 ReportClient 接口去解决
- 该数据不一定需要持久化，可以按照 polaris-server 节点去查询

**Comment by chuntaojun** [comment]
# （WIP）polaris-server 实现统一的主动push机制

目标


建议实现思路

**Comment by chuntaojun** [comment]


**Reply by cjphaha** [reply]
assign me
